fission-server/src/authority.rs

```diff
@@ -0,0 +1,64 @@
+//! Authority struct and functions
+
+use fission_core::authority::key_material::SUPPORTED_KEYS;
+use std::time::{SystemTime, UNIX_EPOCH};
+use ucan::crypto::did::DidParser;
+
+///////////
+// TYPES //
+///////////
+
+#[derive(Debug)]
+/// Represents the authority of an incoming request
+pub struct Authority {
+    /// https://github.com/ucan-wg/ucan-as-bearer-token#21-entry-point
+    pub ucan: ucan::Ucan,
+}
+
+/////////////////////
+// IMPLEMENTATIONS //
+/////////////////////
+
+impl Authority {
+    /// Validate an authority struct
+    pub async fn validate(&self) -> Result<(), String> {
+        let mut did_parser = DidParser::new(SUPPORTED_KEYS);
+        let current_time = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .ok()
+            .map(|t| t.as_secs());
```

Comment:
@zeeshanlakhani This is the most compact way of doing this, right? I can't say `.map(&:as_secs)` or something like that?

Reply:
```rust
SystemTime.now()
// if you apply this with anyhow, which is fine to do through server (vs core)
.duration_since(UNIX_EPOCH)?
.as_secs()
```

Reply:
I actually am trying to use the `?` more, but often, like in this case, I'm getting this error:
![Screenshot 2023-05-12 at 16 10 00](https://github.com/fission-codes/fission-server/assets/296665/4b9f9529-0283-4242-a6ab-6e71b37dbece)
Which doesn't make sense to me.

Reply:
you'd have to use `anyhow::Result` as well. check out the docs, it doesn't specific the error :).